debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/BlockingReader.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.connector.mysql;
+
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+
+import org.apache.kafka.connect.source.SourceRecord;
+
+/**
+ * A component that blocks doing nothing until the connector task is stopped
+ * 
+ * @author Peter Goransson
+ *
+ */
+public class BlockingReader extends AbstractReader {
+
+    private final CountDownLatch latch = new CountDownLatch(1);
+    
+    public BlockingReader(String name, MySqlTaskContext context) {
+        super(name, context);
+    }
+
+    @Override
+    protected void doStart() {
+        logger.info("Connector has completed all of its work but will continue in the running state. It can be shutdown at any time.");
+    }
+
+    @Override
+    protected void doStop() {
+        this.completeSuccessfully();
+        latch.countDown();
```

Comment:
I'd suggest using a try-finally block here, so that the latch is counted down no matter what.

Reply:
All set